### PR TITLE
Auth4

### PR DESCRIPTION
**Pull request by carvilsi**
adding messaging for logIn case that user or password are not right. 
Return to receive array for the permission list of user instead a string

### CODE REVIEW

File: src/WeaverUser.coffee [hunk 1]
```diff
@@ -25,24 +25,26 @@ class WeaverUser
...
       newUserPayload = {newUserCredentials,accessToken}
       coreManager.signUp(newUserPayload)
     )
     
   
   signOff: (currentUsr, user) ->
     coreManager = Weaver.getCoreManager()
-    @current(currentUsr).then((accessToken) ->
+    weaverUser = new WeaverUser()
+    weaverUser.current(currentUsr).then((accessToken) ->
```
Comment (gvdent) [inline src/WeaverUser.coffee]: Can signOff now be static (@signOff) as well?

File: test/WeaverUser.test.coffee [hunk 2]
```diff
@@ -19,80 +22,94 @@ describe 'Weaver User', ->
...
+    weaverUser = new Weaver.User()
+    weaverUser.permission('phoenix').then((res) ->
+      expect(res).to.eql(['read_user', 'create_user', 'delete_user', 'create_role', 'read_role', 'delete_role', 'create_permission', 'read_permission', 'delete_permission', 'read_application', 'create_application', 'delete_application', 'create_directory', 'read_directory', 'delete_directory'])
     )
-    
+  
   it 'should fails when trying to login with non existing user', ->
-    user = new Weaver.User()
-    user.logIn('andromeda','Chains')
+    Weaver.User.logIn(randomstring.generate({length:5,charset:'alphabetic'}),randomstring.generate(7))
```
Comment (gvdent) [inline test/WeaverUser.test.coffee]: In general: refrain from using random anything in unit tests, it damages reproducability. A test containing random may fail or pass depending on the random value, resulting in a horrible heisentestfailure.

File: src/WeaverUser.coffee (hunk 1 above)
Comment (carvilsi) [inline src/WeaverUser.coffee]: 👍

File: test/WeaverUser.test.coffee (hunk 2 above)
Comment (carvilsi) [inline test/WeaverUser.test.coffee]: I'm going to change but for the test environment this should fails always.

**Pull request merged**